Split ZVertexSoA in a run-time sized PortableMultiCollection

fwyzard opened the pull request:
#### PR description:

Split `ZVertexSoA` in two layouts, and wrap them in a multi-layout collection.
Make the ZVertex SoA collections run-time sized, and replace the `ZVertexHost`/`ZVertexDevice` template classes inheriting from `PortableHostMultiCollection`/`PortableDeviceMultiCollection` with type aliases.

Remove versioning info from class description for dictionaries.
Port comments from the CUDA version.

The maximum number of vertices is now a configuration parameter of the `PixelVertexProducerAlpaka` `EDProducer`, while all device-side checks now make use of the various SoA capacity.
The maximum number of vertices is set to
  - `256` for Run 3
  - `512` for Phase 2
  - `32` for heavy ions
These values are probably conservative, and could be further reduced.
The tests use the legacy hard-coded values, taken from the old Run 3 defaults.
    
The documentation in `DataFormats/VertexSoA/README.md` is updated accordingly.

#### PR validation:

Unit tests and workflow `12834.403` pass.

ericcano committed:
Split ZVertexSoA in a PortableMultiCollection

Split ZVertexSoA in two layouts, and wrap them in a multi-layout collection.
Remove versioning info from class description for dictionaries.
Port comments from the CUDA version.

fwyzard committed:
Make the ZVertex SoA collections run-time sized

The maximum number of vertices is now a configuration parameter of the
PixelVertexProducerAlpaka EDProducer, while all device-side checks now
make use of the various SoA capacity.

The tests use hard-coded values, taken from the Run 3 defaults.

The documentation in DataFormats/VertexSoA/README.md is updated
accordingly.

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"7e98653a379e327084b1e84297beedd8e5d4cfa4":{"files":["DQM/SiPixelHeterogeneous/plugins/SiPixelCompareVertices.cc","DQM/SiPixelHeterogeneous/plugins/SiPixelMonitorVertexSoAAlpaka.cc","DataFormats/VertexSoA/interface/ZVertexDevice.h","DataFormats/VertexSoA/interface/ZVertexHost.h","DataFormats/VertexSoA/interface/ZVertexSoA.h","DataFormats/VertexSoA/src/classes.cc","DataFormats/VertexSoA/src/classes_def.xml","DataFormats/VertexSoA/test/alpaka/ZVertexSoA_test.cc","DataFormats/VertexSoA/test/alpaka/ZVertexSoA_test.dev.cc","DataFormats/VertexSoA/test/alpaka/ZVertexSoA_test.h","RecoTauTag/HLTProducers/src/L2TauTagNNProducerAlpaka.cc","RecoTracker/PixelTrackFitting/plugins/PixelTrackDumpAlpaka.cc","RecoTracker/PixelVertexFinding/plugins/PixelVertexProducerFromSoAAlpaka.cc","RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksByDensity.h","RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksDBSCAN.h","RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksIterative.h","RecoTracker/PixelVertexFinding/plugins/alpaka/fitVertices.h","RecoTracker/PixelVertexFinding/plugins/alpaka/sortByPt2.h","RecoTracker/PixelVertexFinding/plugins/alpaka/splitVertices.h","RecoTracker/PixelVertexFinding/plugins/alpaka/vertexFinder.dev.cc","RecoTracker/PixelVertexFinding/plugins/alpaka/vertexFinder.h","RecoTracker/PixelVertexFinding/test/alpaka/VertexFinder_t.dev.cc"],"squashed":false,"time":1725531928},"87219949c28f602493e55c636feeec4f68905701":{"files":["DataFormats/VertexSoA/interface/ZVertexDevice.h","DataFormats/VertexSoA/interface/ZVertexHost.h","DataFormats/VertexSoA/interface/alpaka/ZVertexSoACollection.h","DataFormats/VertexSoA/src/classes.cc","DataFormats/VertexSoA/src/classes_def.xml","DataFormats/VertexSoA/test/alpaka/ZVertexSoA_test.cc","RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksByDensity.h","RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksDBSCAN.h","RecoTracker/PixelVertexFinding/plugins/alpaka/vertexFinder.dev.cc","RecoTracker/PixelVertexFinding/test/alpaka/VertexFinder_t.dev.cc"],"squashed":false,"time":1725532010},"eb8cf5ceb3d807360b9063c1e6015c5ec8956932":{"files":["DataFormats/VertexSoA/interface/ZVertexDevice.h","DataFormats/VertexSoA/interface/ZVertexHost.h","DataFormats/VertexSoA/src/classes.cc","DataFormats/VertexSoA/src/classes_def.xml"],"squashed":false,"time":1725532009}},"emoji":{},"last_seen_sha":"87219949c28f602493e55c636feeec4f68905701","signatures":{}} -->

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-45887/41639

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQM/SiPixelHeterogeneous/plugins/SiPixelCompareVertices.cc  modified in PR(s): #45853
  - File  RecoTracker/PixelVertexFinding/plugins/PixelVertexProducerFromSoAAlpaka.cc  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksByDensity.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksDBSCAN.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksIterative.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/fitVertices.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/sortByPt2.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/splitVertices.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/vertexFinder.dev.cc  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/vertexFinder.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/test/alpaka/VertexFinder_t.dev.cc  modified in PR(s): #45842

cmsbuild commented:
A new Pull Request was created by  @fwyzard for master.

It involves the following packages:

- DQM/SiPixelHeterogeneous (**dqm**)
- DataFormats/VertexSoA (**heterogeneous, reconstruction**)
- RecoTauTag/HLTProducers (**hlt**)
- RecoTracker/PixelTrackFitting (**reconstruction**)
- RecoTracker/PixelVertexFinding (**reconstruction**)


@Martin-Grunewald, @antoniovagnerini, @cmsbuild, @fwyzard, @jfernan2, @makortel, @mandrenguyen, @mmusich, @nothingface0, @rvenditti, @syuvivida, @tjavaid can you please review it and eventually sign? Thanks.
@GiacomoSguazzoni, @JanFSchulte, @VinInn, @VourMa, @azotz, @dgulhan, @felicepantaleo, @fioriNTU, @gpetruc, @idebruyn, @jandrea, @mbluj, @missirol, @mmusich, @mtosi, @rovere, @silviodonato, @threus this is something you requested to watch as well.
@antoniovilela, @mandrenguyen, @rappoccio, @sextonkennedy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-45887/41642

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQM/SiPixelHeterogeneous/plugins/SiPixelCompareVertices.cc  modified in PR(s): #45853
  - File  RecoTracker/PixelVertexFinding/plugins/PixelVertexProducerFromSoAAlpaka.cc  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksByDensity.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksDBSCAN.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksIterative.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/fitVertices.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/sortByPt2.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/splitVertices.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/vertexFinder.dev.cc  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/vertexFinder.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/test/alpaka/VertexFinder_t.dev.cc  modified in PR(s): #45842

cmsbuild commented:
Pull request #45887 was updated. @Martin-Grunewald, @antoniovagnerini, @cmsbuild, @fwyzard, @jfernan2, @makortel, @mandrenguyen, @mmusich, @nothingface0, @rvenditti, @syuvivida, @tjavaid can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-45887/41685

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQM/SiPixelHeterogeneous/plugins/SiPixelCompareVertices.cc  modified in PR(s): #45853
  - File  DataFormats/VertexSoA/interface/ZVertexDefinitions.h  modified in PR(s): #45694
  - File  RecoTracker/PixelVertexFinding/plugins/PixelVertexProducerFromSoAAlpaka.cc  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/PixelVertexProducerAlpaka.cc  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/PixelVertexWorkSpaceSoADeviceAlpaka.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksByDensity.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksDBSCAN.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksIterative.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/fitVertices.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/sortByPt2.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/splitVertices.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/vertexFinder.dev.cc  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/vertexFinder.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/test/alpaka/VertexFinder_t.dev.cc  modified in PR(s): #45842

cmsbuild commented:
Pull request #45887 was updated. @Martin-Grunewald, @antoniovagnerini, @cmsbuild, @fwyzard, @jfernan2, @makortel, @mandrenguyen, @mmusich, @nothingface0, @rvenditti, @syuvivida, @tjavaid can you please check and sign again.

makortel reviewed (commented):
Just to make it clear (or remind), this PR will make all files with the earlier definition of `ZVertexHost` unreadable (which I believe to be fine).

fwyzard committed:
Clean up the pixel vertex kernels

Rename variables for consistency across the pixel vertex reconstruction.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-45887/41686

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQM/SiPixelHeterogeneous/plugins/SiPixelCompareVertices.cc  modified in PR(s): #45853
  - File  DataFormats/VertexSoA/interface/ZVertexDefinitions.h  modified in PR(s): #45694
  - File  RecoTracker/PixelVertexFinding/plugins/PixelVertexProducerFromSoAAlpaka.cc  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/PixelVertexProducerAlpaka.cc  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/PixelVertexWorkSpaceSoADeviceAlpaka.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksByDensity.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksDBSCAN.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksIterative.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/fitVertices.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/sortByPt2.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/splitVertices.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/vertexFinder.dev.cc  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/vertexFinder.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/test/alpaka/VertexFinder_t.dev.cc  modified in PR(s): #45842

cmsbuild commented:
Pull request #45887 was updated. @Martin-Grunewald, @antoniovagnerini, @cmsbuild, @fwyzard, @jfernan2, @makortel, @mandrenguyen, @mmusich, @nothingface0, @rvenditti, @syuvivida, @tjavaid can you please check and sign again.

fwyzard commented:
please test

fwyzard commented:
@AdrianoDee can you double check this, and eventually implement the necessary configuration changes for the Phase 2 workflows ?

cmsbuild commented:
Pull request #45887 was updated. @Martin-Grunewald, @antoniovagnerini, @cmsbuild, @fwyzard, @jfernan2, @makortel, @mandrenguyen, @mmusich, @nothingface0, @rvenditti, @syuvivida, @tjavaid can you please check and sign again.

fwyzard commented:
enable gpu

fwyzard commented:
please test with #45694

cmsbuild commented:
-1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-884194/41345/summary.html
**COMMIT**: 297b861a383a69e55c64c3079f37c50ec4337936
**CMSSW**: CMSSW_14_2_X_2024-09-05-1100/el8_amd64_gcc12
**Additional Tests**: GPU
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/45887/41345/install.sh` to create a dev area with all the needed externals and cmssw changes.

This pull request cannot be automatically merged, could you please rebase it?
You can see the log for git cms-merge-topic here: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-884194/41345/git-merge-result

fwyzard commented:
please test

AdrianoDee commented:
> can you double check this, and eventually implement the necessary configuration changes for the Phase 2 workflows ?

Sure

cmsbuild commented:
+1

**Size**: This PR adds an extra 44KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-884194/41348/summary.html
**COMMIT**: 297b861a383a69e55c64c3079f37c50ec4337936
**CMSSW**: CMSSW_14_2_X_2024-09-05-1100/el8_amd64_gcc12
**Additional Tests**: GPU
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/45887/41348/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [added 1 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-09-05-1100+884194/64397/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 8 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-09-05-1100+884194/64397/validateJR.html)
 - DQMHistoTests: Total files compared: 45
 - DQMHistoTests: Total histograms compared: 3435122
 - DQMHistoTests: [Total failures: 10](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-09-05-1100+884194/64397/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3435092
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 44 files compared)
 - Checked 197 log files, 168 edm output root files, 45 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-09-05-1100+884194/64397/triggerResults): no differences found


## GPU Comparison Summary



**Summary**:
 - You potentially [removed 20 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsGPU/CMSSW_14_2_X_2024-09-05-1100+884194/64383/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 0 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsGPU/CMSSW_14_2_X_2024-09-05-1100+884194/64383/validateJR.html)
 - DQMHistoTests: Total files compared: 6
 - DQMHistoTests: Total histograms compared: 37044
 - DQMHistoTests: [Total failures: 17](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsGPU/CMSSW_14_2_X_2024-09-05-1100+884194/64383/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 37027
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 5 files compared)
 - Checked 20 log files, 25 edm output root files, 6 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsGPU/CMSSW_14_2_X_2024-09-05-1100+884194/64383/triggerResults): no differences found

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-45887/41694

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQM/SiPixelHeterogeneous/plugins/SiPixelCompareVertices.cc  modified in PR(s): #45853
  - File  DataFormats/VertexSoA/interface/ZVertexDefinitions.h  modified in PR(s): #45694
  - File  RecoTracker/PixelVertexFinding/plugins/PixelVertexProducerFromSoAAlpaka.cc  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/PixelVertexProducerAlpaka.cc  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/PixelVertexWorkSpaceSoADeviceAlpaka.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksByDensity.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksDBSCAN.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksIterative.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/fitVertices.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/sortByPt2.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/splitVertices.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/vertexFinder.dev.cc  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/vertexFinder.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/test/alpaka/VertexFinder_t.dev.cc  modified in PR(s): #45842

cmsbuild commented:
Pull request #45887 was updated. @Martin-Grunewald, @antoniovagnerini, @cmsbuild, @fwyzard, @jfernan2, @makortel, @mandrenguyen, @mmusich, @nothingface0, @rvenditti, @syuvivida, @tjavaid can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-45887/41696

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQM/SiPixelHeterogeneous/plugins/SiPixelCompareVertices.cc  modified in PR(s): #45853
  - File  DataFormats/VertexSoA/interface/ZVertexDefinitions.h  modified in PR(s): #45694
  - File  RecoTracker/PixelVertexFinding/plugins/PixelVertexProducerFromSoAAlpaka.cc  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/PixelVertexProducerAlpaka.cc  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/PixelVertexWorkSpaceSoADeviceAlpaka.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksByDensity.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksDBSCAN.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksIterative.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/fitVertices.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/sortByPt2.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/splitVertices.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/vertexFinder.dev.cc  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/vertexFinder.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/test/alpaka/VertexFinder_t.dev.cc  modified in PR(s): #45842

cmsbuild commented:
Pull request #45887 was updated. @Martin-Grunewald, @antoniovagnerini, @cmsbuild, @fwyzard, @jfernan2, @makortel, @mandrenguyen, @mmusich, @nothingface0, @rvenditti, @syuvivida, @tjavaid can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-45887/41697

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQM/SiPixelHeterogeneous/plugins/SiPixelCompareVertices.cc  modified in PR(s): #45853
  - File  DataFormats/VertexSoA/interface/ZVertexDefinitions.h  modified in PR(s): #45694
  - File  RecoTracker/PixelVertexFinding/plugins/PixelVertexProducerFromSoAAlpaka.cc  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/PixelVertexProducerAlpaka.cc  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/PixelVertexWorkSpaceSoADeviceAlpaka.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksByDensity.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksDBSCAN.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/clusterTracksIterative.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/fitVertices.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/sortByPt2.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/splitVertices.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/vertexFinder.dev.cc  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/plugins/alpaka/vertexFinder.h  modified in PR(s): #45842
  - File  RecoTracker/PixelVertexFinding/test/alpaka/VertexFinder_t.dev.cc  modified in PR(s): #45842

cmsbuild commented:
Pull request #45887 was updated. @Martin-Grunewald, @antoniovagnerini, @cmsbuild, @fwyzard, @jfernan2, @makortel, @mandrenguyen, @mmusich, @nothingface0, @rvenditti, @syuvivida, @tjavaid can you please check and sign again.

AdrianoDee committed:
Update the VertexSoA size for Run3, HI, Phase2

AdrianoDee commented on RecoTracker/PixelVertexFinding/plugins/alpaka/PixelVertexProducerAlpaka.cc:
```diff
@@ -61,6 +64,7 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
               conf.getParameter<double>("eps"),
               conf.getParameter<double>("errmax"),
               conf.getParameter<double>("chi2max")),
+        maxVertices_(conf.getParameter<int>("maxVertices")),
```
Ii think we can lower the default (Run3) here to something like `256` (if we want to keep powers of 2, `128` could happen for some very high PU event) given we have

<img width="1099" alt="image" src="https://github.com/user-attachments/assets/ec25d8a1-cb0b-4d9c-97e6-f8a5ee43e3dc">

[in a recent TTbar PU 2024 RelVal](https://tinyurl.com/2ars67bb). Then we could have:

- `512` for Phase2;
- `32` for HIN;

I've put the changes in https://github.com/AdrianoDee/cmssw/commit/a28e5490b62e03ca97a904005ad9277cc332c995.

AdrianoDee replied:
The `1024` was there to be super safe for Phase2. Maybe too much safe.

fwyzard replied:
It doesn't need to be a power of 2 :-)

mmusich commented:
This needs now a rebase (because https://github.com/cms-sw/cmssw/pull/45842 was merged in the meanwhile)

fwyzard committed:
Explicitly use the 1D accelerator

Explicitly use the ALPAKA_ACCELERATOR_NAMESPACE::Acc1D accelerator,
instead of leaving it as a generic template parameter.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-45887/41700

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQM/SiPixelHeterogeneous/plugins/SiPixelCompareVertices.cc  modified in PR(s): #45853

cmsbuild commented:
Pull request #45887 was updated. @Martin-Grunewald, @antoniovagnerini, @cmsbuild, @fwyzard, @jfernan2, @makortel, @mandrenguyen, @mmusich, @nothingface0, @rvenditti, @syuvivida, @tjavaid can you please check and sign again.

fwyzard commented:
> This needs now a rebase (because #45842 was merged in the meanwhile)

OK, I'll come back to this once we have an IB with it.

fwyzard commented:
Rebased on top of a recent IB, and cherry-picked @AdrianoDee's update to the collection sizes.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-45887/41717

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQM/SiPixelHeterogeneous/plugins/SiPixelCompareVertices.cc  modified in PR(s): #45853

cmsbuild commented:
Pull request #45887 was updated. @Martin-Grunewald, @antoniovagnerini, @cmsbuild, @fwyzard, @jfernan2, @makortel, @mandrenguyen, @mmusich, @nothingface0, @rvenditti, @syuvivida, @tjavaid can you please check and sign again.

fwyzard commented:
enable gpu

fwyzard commented:
test parameters:
  - workflows = 12834.403
  - enable_tests = gpu
  - workflows_gpu = 12834.403

fwyzard commented:
please test

cmsbuild commented:
-1

**Failed Tests**: RelVals 
**Size**: This PR adds an extra 52KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-884194/41423/summary.html
**COMMIT**: c7858cbc219453d3d14db91569ca009ea4a54279
**CMSSW**: CMSSW_14_2_X_2024-09-09-1100/el8_amd64_gcc12
**Additional Tests**: GPU
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/45887/41423/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

ValueError: Undefined workflows: 12834.403


## GPU Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - [Reco comparison results: 9 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsGPU/CMSSW_14_2_X_2024-09-09-1100+884194/64470/validateJR.html)
 - DQMHistoTests: Total files compared: 7
 - DQMHistoTests: Total histograms compared: 53003
 - DQMHistoTests: [Total failures: 323](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsGPU/CMSSW_14_2_X_2024-09-09-1100+884194/64470/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 52680
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 6 files compared)
 - Checked 24 log files, 30 edm output root files, 7 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsGPU/CMSSW_14_2_X_2024-09-09-1100+884194/64470/triggerResults): no differences found

AdrianoDee commented:
test parameters:
- enable_test = gpu

AdrianoDee commented:
please test

Something convoluted happened: ˋ12834.403ˋ is now included in the default gpu tests (and indeed has been run). Then it's "not found" as an extra wf since it would need `relval_opt_gpu = -w upgrade`. 🪄

AdrianoDee commented:
please abort

AdrianoDee commented:
please test

fwyzard commented:
can you ask the bot to test it also on cpu ?

AdrianoDee commented:
please abort

AdrianoDee commented:
test parameters:

- workflows = 12834.403
- enable_tests = gpu
- relval_opts = -w upgrade

AdrianoDee commented:
please abort

cmsbuild commented:
+1

**Size**: This PR adds an extra 52KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-884194/41428/summary.html
**COMMIT**: c7858cbc219453d3d14db91569ca009ea4a54279
**CMSSW**: CMSSW_14_2_X_2024-09-09-2300/el8_amd64_gcc12
**Additional Tests**: GPU
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/45887/41428/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [added 2 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-09-09-2300+884194/64473/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 130 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-09-09-2300+884194/64473/validateJR.html)
 - DQMHistoTests: Total files compared: 46
 - DQMHistoTests: Total histograms compared: 3453987
 - DQMHistoTests: [Total failures: 6192](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-09-09-2300+884194/64473/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3447775
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 45 files compared)
 - Checked 201 log files, 173 edm output root files, 46 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-09-09-2300+884194/64473/triggerResults): no differences found


## GPU Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - [Reco comparison results: 0 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsGPU/CMSSW_14_2_X_2024-09-09-2300+884194/64472/validateJR.html)
 - DQMHistoTests: Total files compared: 7
 - DQMHistoTests: Total histograms compared: 53003
 - DQMHistoTests: [Total failures: 244](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsGPU/CMSSW_14_2_X_2024-09-09-2300+884194/64472/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 52759
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 6 files compared)
 - Checked 24 log files, 30 edm output root files, 7 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsGPU/CMSSW_14_2_X_2024-09-09-2300+884194/64472/triggerResults): no differences found

fwyzard commented:
+heterogeneous

mmusich commented:
+hlt
   * (direct) changes in HLT domain are minimal (`RecoTauTag/HLTProducers/src/L2TauTagNNProducerAlpaka.cc`)

fwyzard commented:
kind ping to @cms-sw/dqm-l2 🙏🏻

antoniovagnerini commented:
+1
- spurious differences seen from unstable WF `12434.7`  and `12834.7`   (open issue #39803 )

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @sextonkennedy, @rappoccio, @antoniovilela, @mandrenguyen (and backports should be raised in the release meeting by the corresponding L2)

mandrenguyen commented:
+1

mandrenguyen commented:
cross posting the following comment:
https://github.com/cms-sw/cmssw/pull/46686#issuecomment-2473637407

Is there a failure related to this PR?

makortel commented:
> Is there a failure related to this PR?

Nothing has been reported since the PR was merged, and the IB GPU tests are clean (for the history that is visible). On the other hand, this doesn't prove much because the workflow that failed in https://github.com/cms-sw/cmssw/pull/46686#issuecomment-2473565507, 14949.402, is not tested in IBs.
